### DISCUSSION TITLE
apns-collapse-id not working?

### DISCUSSION THREAD

**Discussion by JunyuKuang** [swift-server-community/APNSwift#129] [Q&A] [answered]
I resend a push notification with same content and apns-collapse-id if Apple server is not responding before timeout. But sometimes the notification I resend is actually delivered successfully and then (unexpectedly) I ended up with duplicated notifications on user's device. 

I'm not sure if it's an iOS bug or APNSwift bug because when I delete one of the duplicated notification from iOS Notification Center, the another one with the same apns-collapse-id gets deleted as well, so iOS appears to recognized them as the same notification, but only when I deleting them.

Has anyone experienced this issue?

**Comment by kylebrowning** [comment, marked as answer]
I just tested it and it worked for me. This is 4 pushes, with two different collapse identifiers
```swift
try await apns.send(notification, pushType: .alert, to: dt, collapseIdentifier: "1234")
        try await apns.send(
            notification, pushType: .alert, to: dt, expiration: expiry, priority: 10, collapseIdentifier: "1234")
        try await apns.send(
            notification, pushType: .alert, to: dt, expiration: expiry, priority: 10, collapseIdentifier: "12345")
        /// Overriden environment
        try await apnsProd.send(aps, to: dt, on: .sandbox, collapseIdentifier: "12345")
```
![Simulator Screen Shot - iPhone 13 Pro Max - 2022-06-11 at 10 50 50](https://user-images.githubusercontent.com/113023/173199201-f2c30061-1b2f-44ba-85bf-c54b0ff93b6f.png)
